Mathlib/CategoryTheory/Limits/Shapes/Countable.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.CategoryTheory.Limits.Final
+import Mathlib.CategoryTheory.Limits.Shapes.FiniteLimits
+import Mathlib.CategoryTheory.Countable
+import Mathlib.Data.Countable.Defs
+/-!
+# Countable limits and colimits
+
+A typeclass for categories with all countable (co)limits.
+
+We also prove that all cofiltered limits over countable preorders are isomorphic to sequential
+limits, see `sequentialFunctor_initial`.
+
+## Projects
+
+* There is a series of `proof_wanted` at the bottom of this file, implying that all cofiltered
+  limits over countable categories are isomorphic to sequential limits.
+
+* Prove the dual result for filtered colimits.
+
+-/
+
+open CategoryTheory Opposite CountableCategory
+
+variable (C : Type*) [Category C] (J : Type*) [Countable J]
+
+namespace CategoryTheory.Limits
+
+/--
+A category has all countable limits if every functor `J ⥤ C` with a `CountableCategory J`
+instance and `J : Type` has a limit.
+-/
+class HasCountableLimits : Prop where
+  /-- `C` has all limits over any type `J` whose objects and morphisms lie in the same universe
+  and which has countably many objects and morphisms-/
+  out (J : Type) [SmallCategory J] [CountableCategory J] : HasLimitsOfShape J C
+
+instance (priority := 100) hasFiniteLimits_of_hasCountableLimits [HasCountableLimits C] :
+    HasFiniteLimits C where
+  out J := HasCountableLimits.out J
+
+instance (priority := 100) hasCountableLimits_of_hasLimits [HasLimits C] :
+    HasCountableLimits C where
+  out := inferInstance
+
+instance [Category J] [CountableCategory J] [HasCountableLimits C] : HasLimitsOfShape J C :=
+  have : HasLimitsOfShape (ObjAsType J) C := HasCountableLimits.out _
+  hasLimitsOfShape_of_equivalence (objAsTypeEquiv J)
+
+/--
+A category has all countable colimits if every functor `J ⥤ C` with a `CountableCategory J`
+instance and `J : Type` has a colimit.
+-/
+class HasCountableColimits : Prop where
+  /-- `C` has all limits over any type `J` whose objects and morphisms lie in the same universe
+  and which has countably many objects and morphisms-/
+  out (J : Type) [SmallCategory J] [CountableCategory J] : HasColimitsOfShape J C
+
+instance (priority := 100) hasFiniteColimits_of_hasCountableColimits [HasCountableColimits C] :
+    HasFiniteColimits C where
+  out J := HasCountableColimits.out J
+
+instance (priority := 100) hasCountableColimits_of_hasColimits [HasColimits C] :
+    HasCountableColimits C where
+  out := inferInstance
+
+instance [Category J] [CountableCategory J] [HasCountableColimits C] : HasColimitsOfShape J C :=
```

Comment:
```suggestion
universe v in
instance [Category.{v} J] [CountableCategory J] [HasCountableColimits C] : HasColimitsOfShape J C :=
```
This is not acutally universe polymorphic. The proof fails if you specify the universe.

Reply:
Ah, I see! I'll fix that in a bit

Reply:
Now it works, sorry it took so long